src/FNA3D_Driver_Vulkan.c

```diff
@@ -2914,32 +2892,24 @@ static uint8_t VULKAN_INTERNAL_BindMemoryForBuffer(
 		{
 			break;
 		}
-		else /* Bind failed, try the next device-local heap */
+		else /* Bind failed, try the next heap */
 		{
 			memoryTypeIndex += 1;
 		}
 	}
 
-	/* Bind failed, try again if originally preferred device local */
-	if (bindResult != 1 && preferDeviceLocal)
+	/* Bind failed, try again without CACHED */
+	if (bindResult != 1)
 	{
 		memoryTypeIndex = 0;
 		requiredMemoryPropertyFlags =
 			VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT |
 			VK_MEMORY_PROPERTY_HOST_COHERENT_BIT;
 
-		/* Follow-up for the warning logged by FindMemoryType */
-		if (!renderer->unifiedMemoryWarning)
-		{
-			FNA3D_LogWarn("No unified memory found, falling back to host memory");
-			renderer->unifiedMemoryWarning = 1;
-		}
-
 		while (VULKAN_INTERNAL_FindBufferMemoryRequirements(
```

Comment:
what happens if we never find any suitable memory? it looks like we fall through to the bottom or we loop forever?

Reply:
memoryTypeIndex is incremented on these calls, so if an alloc fails on a particular memory type it can try again with another memory type. eventually this function will fail and cause the memory bind to fail. at that point the system would be completely out of memory so everything is extremely screwed anyway